src/api/pdc_client_connect.c

```diff
@@ -2376,6 +2376,10 @@ PDC_Client_query_metadata_name_timestep_agg(const char *obj_name, int time_step,
     MPI_Bcast(metadata_server_id, 1, MPI_UINT32_T, 0, PDC_CLIENT_COMM_WORLD_g);
 #else
     ret_value = PDC_Client_query_metadata_name_timestep(obj_name, time_step, out, metadata_server_id);
+    if (ret_value != SUCCEED || NULL == *out) {
+        *out = (pdc_metadata_t *)calloc(1, sizeof(pdc_metadata_t));
```

Comment:
Is this freed somewhere?

Reply:
Yes, it should be freed by PDC_obj_close